woocommerce/woocommerce
# MVP process manifest JSON and convert to WordPress categories/posts

### Changes proposed in this Pull Request:

As a next step in https://github.com/woocommerce/woocommerce/issues/38376 and trying to keep PR's not too large, this introduces some basic functionality to the WooCommerce Docs plugin:

1. A basic script that generates a prototype manifest from a directory consisting of markdown files. To regenerate this example manifest you can run (from woocommerce docs plugin dir) `pnpm generate-manifest "add/wc-docs-manifest"`. The branch name is manually supplied for now to generate raw github links to files that currently only exist in this branch. In `scripts/manifest.json` you can see the example manifest that is generated. This manifest is subject to change and is for now a first prototype for proof of concept.

2. Plugin code to run an action scheduler job that will look through all the manifest URLs that you've added to the plugin via the UI and load those manifests determining Wordpress posts and categories that should be generated based on the manifest. It has some simple logic that checks a hash based ID to determine if the job should re-run or not. If the hash hasn't changed since last time we don't create any new pages. Again this is subject to change, just a suggestion that's been implemented for MVP.

I think the only things left to do for an MVP that this doesn't do are:

* Generate blocks instead of plain HTML
* Parsing of links between docs into full urls?

I will address some of these in follow up PRs to try keep the PR size reasonable.



### How to test the changes in this Pull Request:

1. Have the docs plugin mounted in your dev environment. 
2. Build the docs plugin, via `pnpm start`
3. Now visit the menu item for the plugin in wp-admin (called WooCommerce Docs)
4. Add a new manifest URL via the basic UI, it will need to point to the sample manifest that is included with the plugin. The URL should be like `http://your-dev-hostname/wp-content/plugins/woocommerce-docs/scripts/manifest.json`
5. The action scheduler job runs once a minute after activation of the plugin. Once it has run it should have generated a set of categories and posts with content matching that of the sample docs in `example-docs` folder that is bundled with the plugin currently.

NOTE: If you have trouble getting action scheduler to run actions (sometimes I had trouble with it in dev) you can just run the actions manually via the action UI in ap-admin)
